### ISSUE TITLE
Ubuntu 16.04 - Hunter compilation error

### ISSUE THREAD

**Issue by swkonz** [elucideye/drishti#673]
Hi, I'm attempting to compile directly onto an Android device using the default polly command:
`polly.py --toolchain ${TOOLCHAIN} --config ${CONFIG} --fwd HUNTER_CONFIGURATION_TYPES=${CONFIG} --install --verbose --reconfig --fwd DRISHTI_BUILD_QT=ON`

I'm using the following toolchain:
`export TOOLCHAIN=android-ndk-r10e-api-19-armeabi-v7a-neon-hid-sections`
`export CONFIG=Release`

The log from when the error occurred is below:

[ 75%] Building CXX object src/app/qt/facefilter/CMakeFiles/facefilter.dir/main.cpp.o
cd /home/sean/Desktop/drishti/_builds/android-ndk-r10e-api-19-armeabi-v7a-neon-hid-sections-Release/src/app/qt/facefilter && /home/sean/Desktop/drishti/_ci/android-ndk-r10e/toolchains/arm-linux-androideabi-4.9/prebuilt/linux-x86_64/bin/arm-linux-androideabi-g++ --sysroot=/home/sean/Desktop/drishti/_ci/android-ndk-r10e/platforms/android-19/arch-arm  -DACF_ADD_TO_STRING=1 -DACF_DO_HALF=1 -DANDROID -DDRISHTI_BUILD_MIN_SIZE=1 -DDRISHTI_BUILD_OGLES_GPGPU=1 -DDRISHTI_BUILD_REGRESSION_FIXED_POINT=0 -DDRISHTI_BUILD_REGRESSION_SIMD=0 -DEGL_EGLEXT_PROTOTYPES -DEIGEN_HAS_CXX11_MATH=0 -DGL_GLEXT_PROTOTYPES -DHALF_ENABLE_CPP11_CMATH=0 -DQT_CORE_LIB -DQT_GUI_LIB -DQT_MULTIMEDIA_LIB -DQT_NETWORK_LIB -DQT_NO_DEBUG -DQT_QML_LIB -DQT_QUICK_LIB -DSPDLOG_FMT_EXTERNAL=1 -DXGBOOST_ADD_TO_STRING=1 -DXGBOOST_DO_LEAN=1 -DXGBOOST_SQUEEZE=1 -DXGBOOST_USE_CEREAL=1 -D_USE_MATH_DEFINES -Dfacefilter_EXPORTS -I/home/sean/Desktop/drishti/_builds/android-ndk-r10e-api-19-armeabi-v7a-neon-hid-sections-Release/src/app/qt/facefilter -I/home/sean/Desktop/drishti/src/app/qt/facefilter -I/home/sean/Desktop/drishti/_builds/android-ndk-r10e-api-19-armeabi-v7a-neon-hid-sections-Release/src/app/qt/facefilter/facefilter_autogen/include -isystem /home/sean/.hunter/_Base/428c555/448ccac/f5e0039/Install/include -isystem /home/sean/.hunter/_Base/428c555/448ccac/f5e0039/Install/include/QtQuick -isystem /home/sean/.hunter/_Base/428c555/448ccac/f5e0039/Install/include/QtGui -isystem /home/sean/.hunter/_Base/428c555/448ccac/f5e0039/Install/include/QtCore -isystem /home/sean/.hunter/_Base/428c555/448ccac/f5e0039/Install/./mkspecs/android-g++ -isystem /home/sean/.hunter/_Base/428c555/448ccac/f5e0039/Install/include/QtQml -isystem /home/sean/.hunter/_Base/428c555/448ccac/f5e0039/Install/include/QtNetwork -isystem /home/sean/.hunter/_Base/428c555/448ccac/f5e0039/Install/include/QtMultimedia -I/home/sean/Desktop/drishti/src/lib -isystem /home/sean/.hunter/_Base/428c555/448ccac/f5e0039/Install/sdk/native/jni/include -isystem /home/sean/.hunter/_Base/428c555/448ccac/f5e0039/Install/sdk/native/jni/include/opencv -isystem /home/sean/.hunter/_Base/428c555/448ccac/f5e0039/Install/include/eigen3 -isystem /home/sean/.hunter/_Base/428c555/448ccac/f5e0039/Install/include/eigen3/unsupported -isystem /home/sean/Desktop/drishti/_ci/android-ndk-r10e/sources/cxx-stl/gnu-libstdc++/4.9/include -isystem /home/sean/Desktop/drishti/_ci/android-ndk-r10e/sources/cxx-stl/gnu-libstdc++/4.9/libs/armeabi-v7a/include -isystem /home/sean/Desktop/drishti/_ci/android-ndk-r10e/sources/cxx-stl/gnu-libstdc++/4.9/include/backward -isystem /home/sean/Desktop/drishti/_ci/android-ndk-r10e/platforms/android-19/arch-arm/usr/include  -fvisibility-inlines-hidden -fvisibility=hidden -fdata-sections -ffunction-sections -std=c++11   -march=armv7-a -marm -mfpu=neon -mfloat-abi=softfp -funwind-tables -no-canonical-prefixes -fexceptions -frtti -Wno-c++11-narrowing -Wno-unused-parameter -Wno-unused-local-typedefs -Wno-unknown-attributes -Wno-shorten-64-to-32 -O3 -DNDEBUG -fPIC   -fPIC -std=gnu++11 -o CMakeFiles/facefilter.dir/main.cpp.o -c /home/sean/Desktop/drishti/src/app/qt/facefilter/main.cpp
In file included from /home/sean/.hunter/_Base/428c555/448ccac/f5e0039/Install/include/QtGui/QOpenGLExtraFunctions:1:0,
                 from /home/sean/Desktop/drishti/src/app/qt/facefilter/main.cpp:54:
/home/sean/.hunter/_Base/428c555/448ccac/f5e0039/Install/include/QtGui/qopenglextrafunctions.h:313:5: error: 'GLsync' does not name a type
     GLsync glFenceSync(GLenum condition, GLbitfield flags);
     ^
/home/sean/.hunter/_Base/428c555/448ccac/f5e0039/Install/include/QtGui/qopenglextrafunctions.h:314:24: error: 'GLsync' has not been declared
     GLboolean glIsSync(GLsync sync);
                        ^
/home/sean/.hunter/_Base/428c555/448ccac/f5e0039/Install/include/QtGui/qopenglextrafunctions.h:315:23: error: 'GLsync' has not been declared
     void glDeleteSync(GLsync sync);
                       ^
/home/sean/.hunter/_Base/428c555/448ccac/f5e0039/Install/include/QtGui/qopenglextrafunctions.h:316:29: error: 'GLsync' has not been declared
     GLenum glClientWaitSync(GLsync sync, GLbitfield flags, GLuint64 timeout);
                             ^
/home/sean/.hunter/_Base/428c555/448ccac/f5e0039/Install/include/QtGui/qopenglextrafunctions.h:316:60: error: 'GLuint64' has not been declared
     GLenum glClientWaitSync(GLsync sync, GLbitfield flags, GLuint64 timeout);
                                                            ^
/home/sean/.hunter/_Base/428c555/448ccac/f5e0039/Install/include/QtGui/qopenglextrafunctions.h:317:21: error: 'GLsync' has not been declared
     void glWaitSync(GLsync sync, GLbitfield flags, GLuint64 timeout);
                     ^
/home/sean/.hunter/_Base/428c555/448ccac/f5e0039/Install/include/QtGui/qopenglextrafunctions.h:317:52: error: 'GLuint64' has not been declared
     void glWaitSync(GLsync sync, GLbitfield flags, GLuint64 timeout);
                                                    ^
/home/sean/.hunter/_Base/428c555/448ccac/f5e0039/Install/include/QtGui/qopenglextrafunctions.h:318:40: error: 'GLint64' has not been declared
     void glGetInteger64v(GLenum pname, GLint64 *data);
                                        ^
/home/sean/.hunter/_Base/428c555/448ccac/f5e0039/Install/include/QtGui/qopenglextrafunctions.h:319:22: error: 'GLsync' has not been declared
     void glGetSynciv(GLsync sync, GLenum pname, GLsizei bufSize, GLsizei *length, GLint *values);
                      ^
/home/sean/.hunter/_Base/428c555/448ccac/f5e0039/Install/include/QtGui/qopenglextrafunctions.h:320:57: error: 'GLint64' has not been declared
     void glGetInteger64i_v(GLenum target, GLuint index, GLint64 *data);
                                                         ^
/home/sean/.hunter/_Base/428c555/448ccac/f5e0039/Install/include/QtGui/qopenglextrafunctions.h:321:64: error: 'GLint64' has not been declared
     void glGetBufferParameteri64v(GLenum target, GLenum pname, GLint64 *params);
                                                                ^
In file included from /home/sean/.hunter/_Base/428c555/448ccac/f5e0039/Install/include/QtGui/qopenglfunctions.h:54:0,
                 from /home/sean/.hunter/_Base/428c555/448ccac/f5e0039/Install/include/QtGui/QOpenGLFunctions:1,
                 from /home/sean/Desktop/drishti/src/app/qt/facefilter/VideoFilterRunnable.hpp:46,
                 from /home/sean/Desktop/drishti/src/app/qt/facefilter/main.cpp:46:
/home/sean/.hunter/_Base/428c555/448ccac/f5e0039/Install/include/QtGui/qopenglextrafunctions.h:619:5: error: expected identifier before '*' token
     QT_OPENGL_DECLARE(QT_OPENGL_EXTRA_FUNCTIONS)
     ^
In file included from /home/sean/.hunter/_Base/428c555/448ccac/f5e0039/Install/include/QtGui/QOpenGLExtraFunctions:1:0,
                 from /home/sean/Desktop/drishti/src/app/qt/facefilter/main.cpp:54:
/home/sean/.hunter/_Base/428c555/448ccac/f5e0039/Install/include/QtGui/qopenglextrafunctions.h:619:5: error: 'GLsync' declared as function returning a function
     QT_OPENGL_DECLARE(QT_OPENGL_EXTRA_FUNCTIONS)
     ^
/home/sean/.hunter/_Base/428c555/448ccac/f5e0039/Install/include/QtGui/qopenglextrafunctions.h:619:5: error: 'GLsync' has not been declared
     QT_OPENGL_DECLARE(QT_OPENGL_EXTRA_FUNCTIONS)
     ^
/home/sean/.hunter/_Base/428c555/448ccac/f5e0039/Install/include/QtGui/qopenglextrafunctions.h:619:5: error: 'GLsync' has not been declared
     QT_OPENGL_DECLARE(QT_OPENGL_EXTRA_FUNCTIONS)
     ^
/home/sean/.hunter/_Base/428c555/448ccac/f5e0039/Install/include/QtGui/qopenglextrafunctions.h:619:5: error: 'GLsync' has not been declared
     QT_OPENGL_DECLARE(QT_OPENGL_EXTRA_FUNCTIONS)
     ^
/home/sean/.hunter/_Base/428c555/448ccac/f5e0039/Install/include/QtGui/qopenglextrafunctions.h:619:5: error: 'GLuint64' has not been declared
     QT_OPENGL_DECLARE(QT_OPENGL_EXTRA_FUNCTIONS)
     ^
/home/sean/.hunter/_Base/428c555/448ccac/f5e0039/Install/include/QtGui/qopenglextrafunctions.h:619:5: error: 'GLsync' has not been declared
     QT_OPENGL_DECLARE(QT_OPENGL_EXTRA_FUNCTIONS)
     ^
/home/sean/.hunter/_Base/428c555/448ccac/f5e0039/Install/include/QtGui/qopenglextrafunctions.h:619:5: error: 'GLuint64' has not been declared
     QT_OPENGL_DECLARE(QT_OPENGL_EXTRA_FUNCTIONS)
     ^
/home/sean/.hunter/_Base/428c555/448ccac/f5e0039/Install/include/QtGui/qopenglextrafunctions.h:619:5: error: 'GLint64' has not been declared
     QT_OPENGL_DECLARE(QT_OPENGL_EXTRA_FUNCTIONS)
     ^
/home/sean/.hunter/_Base/428c555/448ccac/f5e0039/Install/include/QtGui/qopenglextrafunctions.h:619:5: error: 'GLsync' has not been declared
     QT_OPENGL_DECLARE(QT_OPENGL_EXTRA_FUNCTIONS)
     ^
/home/sean/.hunter/_Base/428c555/448ccac/f5e0039/Install/include/QtGui/qopenglextrafunctions.h:619:5: error: 'GLint64' has not been declared
     QT_OPENGL_DECLARE(QT_OPENGL_EXTRA_FUNCTIONS)
     ^
/home/sean/.hunter/_Base/428c555/448ccac/f5e0039/Install/include/QtGui/qopenglextrafunctions.h:619:5: error: 'GLint64' has not been declared
     QT_OPENGL_DECLARE(QT_OPENGL_EXTRA_FUNCTIONS)
     ^
/home/sean/.hunter/_Base/428c555/448ccac/f5e0039/Install/include/QtGui/qopenglextrafunctions.h:720:55: error: 'QOpenGLExtraFunctions::glClientWaitSync' declared as an 'inline' variable
 inline GLenum QOpenGLExtraFunctions::glClientWaitSync(GLsync sync, GLbitfield flags, GLuint64 timeout)
                                                       ^
/home/sean/.hunter/_Base/428c555/448ccac/f5e0039/Install/include/QtGui/qopenglextrafunctions.h:720:55: error: 'GLenum QOpenGLExtraFunctions::glClientWaitSync' is not a static member of 'class QOpenGLExtraFunctions'
/home/sean/.hunter/_Base/428c555/448ccac/f5e0039/Install/include/QtGui/qopenglextrafunctions.h:720:55: error: 'GLsync' was not declared in this scope
/home/sean/.hunter/_Base/428c555/448ccac/f5e0039/Install/include/QtGui/qopenglextrafunctions.h:720:79: error: expected primary-expression before 'flags'
 inline GLenum QOpenGLExtraFunctions::glClientWaitSync(GLsync sync, GLbitfield flags, GLuint64 timeout)
                                                                               ^
/home/sean/.hunter/_Base/428c555/448ccac/f5e0039/Install/include/QtGui/qopenglextrafunctions.h:720:86: error: 'GLuint64' was not declared in this scope
 inline GLenum QOpenGLExtraFunctions::glClientWaitSync(GLsync sync, GLbitfield flags, GLuint64 timeout)
                                                                                      ^
/home/sean/.hunter/_Base/428c555/448ccac/f5e0039/Install/include/QtGui/qopenglextrafunctions.h:720:102: error: expression list treated as compound expression in initializer [-fpermissive]
 inline GLenum QOpenGLExtraFunctions::glClientWaitSync(GLsync sync, GLbitfield flags, GLuint64 timeout)
                                                                                                      ^
/home/sean/.hunter/_Base/428c555/448ccac/f5e0039/Install/include/QtGui/qopenglextrafunctions.h:721:1: error: expected ',' or ';' before '{' token
 {
 ^
In file included from /home/sean/Desktop/drishti/src/app/qt/facefilter/main.cpp:60:0:
/home/sean/Desktop/drishti/src/app/qt/facefilter/VideoFilter.hpp:42:20: error: expected declaration before end of line
     Q_OBJECT
                    ^
cc1plus: warning: unrecognized command line option "-Wno-shorten-64-to-32"
cc1plus: warning: unrecognized command line option "-Wno-unknown-attributes"
cc1plus: warning: unrecognized command line option "-Wno-c++11-narrowing"
make[2]: *** [src/app/qt/facefilter/CMakeFiles/facefilter.dir/main.cpp.o] Error 1
src/app/qt/facefilter/CMakeFiles/facefilter.dir/build.make:77: recipe for target 'src/app/qt/facefilter/CMakeFiles/facefilter.dir/main.cpp.o' failed
make[2]: Leaving directory '/home/sean/Desktop/drishti/_builds/android-ndk-r10e-api-19-armeabi-v7a-neon-hid-sections-Release'
CMakeFiles/Makefile2:582: recipe for target 'src/app/qt/facefilter/CMakeFiles/facefilter.dir/all' failed
make[1]: Leaving directory '/home/sean/Desktop/drishti/_builds/android-ndk-r10e-api-19-armeabi-v7a-neon-hid-sections-Release'
Makefile:154: recipe for target 'all' failed
make[1]: *** [src/app/qt/facefilter/CMakeFiles/facefilter.dir/all] Error 2
make: *** [all] Error 2

From what I can tell from the log, it seems like the error is coming from hunter where GLSync is not defined as a type. I'd like to directly compile and load the app onto a device. Any idea what might be causing this error? I've also added my device info to the facefilter.json file:

`    "0x1115fbe111eb1503": {
        "description": "Android phone",

        "sensor": {
            "intrinsic": {
                "size": {
                    "width": 640,
                    "height": 480
                },
                "principal": {
                    "x": 320.0,
                    "y": 240.0
                },
                "focal_length_x": 1000.0
            }
        }
    }`

**Comment by headupinclouds** [comment]
> From what I can tell from the log, it seems like the error is coming from hunter where GLSync is not defined as a type

I ran a build and was able to reproduce the issue.  I can take al look more this weekend.  I don't think it is related to Hunter.  The Qt release drishti is using is already not building on iOS 11.3.  Given the rate of iOS and Android updates, I think keeping the Qt package updated will be fairly time consuming.  We've decided to switch to native iOS/Android instead.  Since the issue is happening at the application level, it might be fixable.  I can take a look this weekend.
